tempesta_fw/tls.c

```diff
@@ -530,8 +539,20 @@ tfw_cfgop_cleanup_ssl_certificate_key(TfwCfgSpec *cs)
 static int
 tfw_tls_cfgend(void)
 {
-	if ((tfw_tls.crt.version && !tfw_tls.key.pk_ctx) ||
-	    (!tfw_tls.crt.version && tfw_tls.key.pk_ctx)) {
+	if (!(tfw_tls_cgf & TFW_TLS_CFG_F_REQUIRED)) {
+		if (tfw_tls_cgf & TFW_TLS_CFG_F_REQUIRED)
+			TFW_WARN_NL("TLS: no HTTPS listener,"
+				    " configuration ignored\n");
+		return 0;
+	}
+
+	if (!(tfw_tls_cgf & TFW_TLS_CFG_F_CERT)) {
+		TFW_ERR_NL("TLS: please spcify a certificate with"
+			   " tls_certificate configuration option\n");
+		return -EINVAL;
+	}
+
+	if ((tfw_tls_cgf & TFW_TLS_CFG_M_ALL) != TFW_TLS_CFG_M_ALL) {
```

Comment:
It looks not very user friendly. If the certificate is missing when Tempesta says, which exactly directive is missed. But if the key is missed, then it just says that `pair is incomplete`. It'll be better to make the next warning:
```
TFW_ERR_NL("TLS: please specify a key with 'tls_certificate_key' configuration option\n");
```